core/src/main/java/com/amplifyframework/auth/AuthCodeDeliveryDetails.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.auth;
+
+public final class AuthCodeDeliveryDetails {
+    private String destination;
+    private String deliveryMedium;
+    private String attributeName;
+
+    public AuthCodeDeliveryDetails(String destination, String deliveryMedium, String attributeName) {
+        this.destination = destination;
+        this.deliveryMedium = deliveryMedium;
```

Comment:
Can this be an enum? Which are the different delivery mediums? Are they known to us? Do we only support a constrained set of them?

Reply:
The answers to your last two questions are both no since AWSMobileClient just returns it as a string. But I guess we could do an enum with the methods we currently know and an unknown value and add more values as needed.

Reply:
Jithin spec'd out an enum for this - you could check the iOS PR / sync with him.

Reply:
Email and SMS are the two values added in iOS which are also supported by [Cognito](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_CodeDeliveryDetailsType.html#CognitoUserPools-Type-CodeDeliveryDetailsType-Destination).

Reply:
Ah yeah whoops - I forgot about adding this - will do.